### DISCUSSION TITLE
getting method-specific function code

### DISCUSSION THREAD

**Discussion by rkb965** [posit-conf-2023/vetiver#30] [Q&A] [answered]
****
Hi -- I want to see how the vetiver description is created -- can someone please help me with the correct syntax? I have found many things that it is not, including:
`getMethod("vetiver_model", "vetiver_create_description")`

Thank you!

**Comment by isabelizimm** [comment, marked as answer]
There is a section for [developer functions](https://rstudio.github.io/vetiver-r/reference/?_gl=1*1y58tlp*_ga*ODA3OTgwOTk5LjE2NzQ3NzM0MDg.*_ga_2C0WZ1JHG0*MTY5NTA1Mzg1My43OS4xLjE2OTUwNTM5MTEuMC4wLjA.#developer-functions) in the vetiver for R documentation! The specifics will depend on the type of model.

**Comment by rkb965** [comment]
Thanks! This is what I was aiming for:
`getS3method("vetiver_create_description","ranger")`

though it still seems magical that it included ` using 4 features` given that the result of ^ is simply:

```
> getS3method("vetiver_create_description","ranger")
function (model) 
{
    glue("A ranger {tolower(model$forest$treetype)} model")
}
<bytecode: 0x5629055af298>
<environment: namespace:vetiver>
```

🤯

**Reply by rkb965** [reply]
Hmm... I thought maybe it was in the print method but not the description? but no 

`getS3method("print","vetiver_model")`

```
function (x, ...) 
{
    cat(format(x), sep = "\n")
    invisible(x)
}
<bytecode: 0x5628f4dc6678>
<environment: namespace:vetiver>
```

**Comment by juliasilge** [comment]
There are two pieces that go into what you see when you print out the model. The first is the `vetiver_create_description` method for the different model classes, like this:
https://github.com/rstudio/vetiver-r/blob/fc8b34980553bac403dc43385dedeae18cbfec36/R/tidymodels.R#L3-L6

And then the second comes from the `format` method for a vetiver model:
https://github.com/rstudio/vetiver-r/blob/fc8b34980553bac403dc43385dedeae18cbfec36/R/vetiver-model.R#L114-L124
